pkg/gateway/gateway.go

```diff
@@ -57,7 +57,7 @@ type gateway struct {
 
 	ready          atomic.Value
 	lastChangefeed atomic.Value //time.Time
-	mu             sync.RWMutex
+	mu             *sync.RWMutex
```

Comment:
> I'd say a good rule of thumb is to have functions on ptr for modifying a struct, and on the struct itself when it is read only. (here it is read only, since initial_state==final_state)

But we do mutate the receiver here: we need ot modify mutex state from these methods.

Moving away from a pointer receiver also means that:

1. We copy values when we call a method wih value receiver. This is unnecessary in most of the cases (
2. It is easy to introduce a nasty bug when you want modify something in the instance of `gateway`, but you actually modify a copy. Chances of this happening are probably the same as chances of modifying something we don't want to modify. Maybe even higher. For example, I can definitely see someone comming in a month or so to this code and modifying `mu *sync.RWMutex` back to `mu sync.RWMutex` (command pattern as it is usable from empty value) without noticing that it is used in methods with value receivers.

BTW, [here](https://github.com/golang/go/wiki/CodeReviewComments#receiver-type) is some good advice, in my opinion. We do not follow it strictly, but I think we de facto default to pointers.

Anyway - this seems out of scope of this PR. If you think that we should continue this discussion - please submit a separate PR for this with a link to this thread.